patches/api/0418-Add-player-death-screen-message-API.patch

```diff
@@ -0,0 +1,63 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: dawon <dawon@dawon.eu>
+Date: Fri, 24 Feb 2023 15:20:47 +0100
+Subject: [PATCH] Add player death screen message API
+
+
+diff --git a/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java b/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
+index d5eeb37d331e205826aa6199bd481cb0db40d721..182ae2d4220d6dec7a7a6a45eba05188fd095266 100644
+--- a/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
++++ b/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
+@@ -21,6 +21,7 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+     private boolean keepInventory = false;
+     // Paper start
+     private boolean doExpDrop;
++    private net.kyori.adventure.text.Component adventure$deathScreenMessage;
+ 
+     public PlayerDeathEvent(@NotNull final Player player, @NotNull final List<ItemStack> drops, final int droppedExp, @Nullable final net.kyori.adventure.text.Component adventure$deathMessage) {
+         this(player, drops, droppedExp, 0, adventure$deathMessage, null);
+@@ -42,6 +43,7 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+         this.deathMessage = deathMessage;
+         this.adventure$deathMessage = adventure$deathMessage;
+         this.doExpDrop = doExpDrop;
++        this.adventure$deathScreenMessage = null;
+     }
+     // Paper end
+ 
+@@ -67,6 +69,7 @@ public class PlayerDeathEvent extends EntityDeathEvent {
+         this.newLevel = newLevel;
+         this.deathMessage = deathMessage;
+         this.adventure$deathMessage = deathMessage != null ? net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().deserialize(deathMessage) : null; // Paper
++        adventure$deathScreenMessage = null; // Paper
```

Comment:
Should this really be null ? Especially when the death message is initialised to its correct value here.
I feel like this should be equal to the death message.

Reply:
You are right, this is probably a more intuitive approach for the developers, thank you, fixed!